### PR TITLE
Add signal loss optimization to RMSE criteria

### PR DESCRIPTION
**Pull request by 4rnaudB**
## Description
After some experiments, it was determined that RMSE might be better suited for shimming in some situations. This PR implements signal recovery optimization for the RMSE criteria and refactors the code to reduce the number of inputs needed.

### CODE REVIEW

File: shimming-toolbox/shimmingtoolbox/cli/b0shim.py [hunk 1]
```diff
@@ -227,9 +230,18 @@ def dynamic(fname_fmap, fname_anat, fname_mask_anat, method, opt_criteria, slice
     with open(fname_anat_json) as json_file:
         json_anat_data = json.load(json_file)
 
-    # Get the EPI echo time if optimization criteria is grad
-    if opt_criteria == 'grad':
+    # Get the EPI echo time and set signal recovery optimizer criteria if w signal loss is set
+    if w_signal_loss or w_signal_loss_xy:
+        if opt_criteria != 'mse':
```
Comment (po09i) [inline shimming-toolbox/shimmingtoolbox/cli/b0shim.py]: rmse as well?

File: shimming-toolbox/shimmingtoolbox/optimizer/lsq_optimizer.py [hunk 2]
```diff
@@ -333,11 +357,7 @@ def _get_currents(self, unshimmed_vec, coil_mat, currents_0):
             # scipy minimize expects the return value of the residual function to be ~10^0 to 10^1
             # --> aiming for 1 then optimizing will lower that. We are using an initial guess of 0s so that the
             # regularization on the currents has no affect on the output stability factor.
-            if self.opt_criteria == 'mse':
-                stability_factor = self._initial_guess_mse(self._initial_guess_zeros(), unshimmed_vec,
-                                                           np.zeros_like(coil_mat),
-                                                           factor=1)
-            elif self.opt_criteria == 'grad':
+            if 'mse' in self.opt_criteria:
```
Comment (po09i) [inline shimming-toolbox/shimmingtoolbox/optimizer/lsq_optimizer.py]: This will also match self.opt_criteria == "rmse"

File: shimming-toolbox/test/cli/test_cli_b0shim.py
Comment (po09i) [inline shimming-toolbox/test/cli/test_cli_b0shim.py:1]: Maybe add a test using signal recovery and "rmse"

Comment (4rnaudB) [reply shimming-toolbox/test/cli/test_cli_b0shim.py:1]: added in [2ae6977](https://github.com/shimming-toolbox/shimming-toolbox/pull/573/commits/2ae6977d3f96b0472bc2adb6546a6d23edca160d)

File: shimming-toolbox/shimmingtoolbox/optimizer/lsq_optimizer.py (hunk 2 above)
Comment (4rnaudB) [reply shimming-toolbox/shimmingtoolbox/optimizer/lsq_optimizer.py]: fixed in [2ae6977](https://github.com/shimming-toolbox/shimming-toolbox/pull/573/commits/2ae6977d3f96b0472bc2adb6546a6d23edca160d)

File: shimming-toolbox/shimmingtoolbox/cli/b0shim.py (hunk 1 above)
Comment (4rnaudB) [reply shimming-toolbox/shimmingtoolbox/cli/b0shim.py]: fixed in [2ae6977](https://github.com/shimming-toolbox/shimming-toolbox/pull/573/commits/2ae6977d3f96b0472bc2adb6546a6d23edca160d)

File: fsleyes-plugin-shimming-toolbox/fsleyes_plugin_shimming_toolbox/tabs/b0shim_tab.py [hunk 3]
```diff
@@ -290,7 +290,8 @@ def create_sizer_dynamic_shim(self, metadata=None):
                 "default_text": "0.01",
             },
         ]
-        component_slice_w_sig_loss = InputComponent(self, weighting_signal_loss_metadata, cli=dynamic_cli)
+        component_slice_w_sig_loss_mse = InputComponent(self, weighting_signal_loss_metadata, cli=dynamic_cli)
+        component_slice_w_sig_loss_rmse = InputComponent(self, weighting_signal_loss_metadata, cli=dynamic_cli)
```
Comment (po09i) [inline fsleyes-plugin-shimming-toolbox/fsleyes_plugin_shimming_toolbox/tabs/b0shim_tab.py]: Maybe have a default of 10 for rmse if that seems to be the optimal value.

Comment (4rnaudB) [reply fsleyes-plugin-shimming-toolbox/fsleyes_plugin_shimming_toolbox/tabs/b0shim_tab.py]: Fixed in [b510cda](https://github.com/shimming-toolbox/shimming-toolbox/pull/573/commits/b510cda010760ed702cbeab9611648debe52427f)

**Review by po09i — APPROVED** [review]
